grails-app/controllers/org/transmartproject/rest/HighDimController.groovy

```diff
@@ -48,39 +54,82 @@ class HighDimController {
     @Resource
     StudyLoadingService studyLoadingServiceProxy
 
-    def show(String dataType, String projection) {
-
-        if (dataType == null) {
-            index()
-        } else {
-            exportData(dataType, projection)
+    def index() {
+        if (params.dataType) {
+            // backwards compatibility
+            // preferred to use /highdim/<data type> for download
+            download params.dataType
+            return
         }
+
+        String conceptKey = getConceptKey(params.conceptId)
+        OntologyTerm concept = conceptsResourceService.getByKey(conceptKey)
+        String conceptLink = studyLoadingServiceProxy.getOntologyTermUrl(concept)
+        String selfLink = HighDimSummarySerializationHelper.getHighDimIndexUrl(conceptLink)
+
+        respond wrapList(getHighDimSummaries(concept), selfLink)
     }
 
-    private void exportData(String dataType, String projection) {
+    def download(String dataType) {
+        assert dataType != null // ensured by mapping
+        def assayConstraintsSpec = processConstraintsJson params.assayConstraints
+        def dataConstraintsSpec = processConstraintsJson params.dataConstraints
+
         String conceptKey = getConceptKey(params.conceptId)
-        OutputStream out = response.outputStream
-        response.contentType =  'application/octet-stream'
+        OutputStream out = new LazyOutputStreamDecorator(
```

Comment:
If we call response.outputStream at this point, we won't be able to send an error response anymore because the headers will be committed. But `highDimDataService.write` can throw. Maybe this could be rearranged better, with the parts that can fail moved before, but I didn't look into it. All I noticed was that when there was an error I'd get a second exception masking the first.